pandas/tests/test_groupby.py

```diff
@@ -248,12 +248,12 @@ def test_nth(self):
 
         assert_frame_equal(g.nth(0), df.iloc[[0, 2]].set_index('A'))
         assert_frame_equal(g.nth(1), df.iloc[[1]].set_index('A'))
-        assert_frame_equal(g.nth(2), df.loc[[],['B']])
+        assert_frame_equal(g.nth(2), df.loc[[]].set_index('A'))
         assert_frame_equal(g.nth(-1), df.iloc[[1, 2]].set_index('A'))
         assert_frame_equal(g.nth(-2), df.iloc[[0]].set_index('A'))
-        assert_frame_equal(g.nth(-3), df.loc[[],['B']])
-        assert_series_equal(g.B.nth(0), df.B.iloc[[0, 2]])
-        assert_series_equal(g.B.nth(1), df.B.iloc[[1]])
+        assert_frame_equal(g.nth(-3), df.loc[[]].set_index('A'))
+        assert_series_equal(g.B.nth(0), df.set_index('A').B.iloc[[0, 2]])
```

Comment:
@jreback i am not sure what u want me to put in what's new; such things would be much more efficient if u take care of it urself. even if i do it, it may not be what u had in mind, and then we have to keep going back and forth.

the other thing is that this whole index and dropna is broken; in addition to https://github.com/pydata/pandas/issues/11038 there is this:

```python
>>> df
  1st  2nd
0   a    0
1   a    1
2   a    2
3   b  NaN
4   b    4
>>> df.groupby('1st')['2nd'].nth(0, dropna=False)
0     0
3   NaN
Name: 2nd, dtype: float64
>>> df.groupby('1st')['2nd'].nth(0, dropna=True)
1st
a   NaN
b   NaN
Name: 2nd, dtype: float64
```

so not sure if there is a point in documenting it when it is not working.